project/BinaryIncompatibilities.scala

```diff
@@ -16,6 +16,16 @@ object BinaryIncompatibilities {
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.scalajs.ir.Types.BoxedClassToPrimType"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.scalajs.ir.Types.PrimTypeToBoxedClass"),
 
+    // !!! Breaking I guess ... we used to leak public things out of a `case class` with a private[ir] constructor
```

Comment:
:scream: the visibility of the case class constructor not being inherited in the synthetic members definitely violates the principle of least surprise. Or were all of these `private[ir]` before? (in which case, I'm not sure it is necessary to do the private + throw dance).

Reply:
No they were truly public. 😱 I tested. They were accessible from another package.